tdms/tests/system/utils.py

```diff
@@ -100,3 +91,53 @@ def relative_mean_squared_difference(a: np.ndarray, b: np.ndarray) -> float:
         return 0.0  # Prevent division by zero for tensors filled with zeros
 
     return np.mean(np.square(a - b)) / max(mean_sq_a, mean_sq_b)
+
+
+def work_in_zipped_dir(zip_path: Path):
+    """
+    Function decorator to extract some data from a compressed folder into a
+    directory, run the function, change directories to the initial working
+    directory and then delete the generated folder.
+    """
+
+    if not str(zip_path).endswith(".zip"):
+        raise ValueError(f"Cannot work in {zip_path}. Not a .zip directory")
+
+    def func_decorator(func):
+        @wraps(func)
+        def wrapped_function(*args, **kwargs):
+
+            cwd = os.getcwd()
+            dir_path = Path(cwd, zip_path.stem)
+
+            with ZipFile(zip_path, "r") as zip_folder:
+                zip_folder.extractall(dir_path)
+
+            os.chdir(dir_path)
+
+            try:
+                result = func(*args, **kwargs)
+
+            finally:
+                os.chdir(cwd)
+                shutil.rmtree(dir_path)
+
+            return result
+
+        return wrapped_function
+
+    return func_decorator
+
+
+def run_tdms(input_filename: str, output_filename: str) -> None:
```

Comment:
Curiosity No. 2 (not really important): is `None` the type of `None`?  Or I misunderstand that the `->` syntax indicates the _type_ of the output of the function?

Reply:
Interesting.. the `->` should be the type of the output but it's I believe convention to have `None` for a void function

```python
>>> type(None)
<class 'NoneType'>
```

but it can't be instantiated – I guess having only one None is a good thing